OpenRA.Mods.Common/Pathfinder/PathGraph.cs

```diff
@@ -18,100 +18,52 @@
 namespace OpenRA.Mods.Common.Pathfinder
 {
 	/// <summary>
-	/// Represents a graph with nodes and edges
+	/// A dense pathfinding graph that supports a search over all cells within a map.
+	/// It implements the ability to cost and get connections for cells, and supports <see cref="ICustomMovementLayer"/>.
 	/// </summary>
-	/// <typeparam name="T">The type of node used in the graph</typeparam>
-	public interface IGraph<T> : IDisposable
-	{
-		/// <summary>
-		/// Gets all the Connections for a given node in the graph
-		/// </summary>
-		List<GraphConnection> GetConnections(CPos position);
-
-		/// <summary>
-		/// Retrieves an object given a node in the graph
-		/// </summary>
-		T this[CPos pos] { get; set; }
-
-		Func<CPos, bool> CustomBlock { get; set; }
-
-		Func<CPos, int> CustomCost { get; set; }
-
-		int LaneBias { get; set; }
-
-		bool InReverse { get; set; }
-
-		Actor IgnoreActor { get; set; }
-
-		World World { get; }
-
-		Actor Actor { get; }
-	}
-
-	public readonly struct GraphConnection
-	{
-		public static readonly CostComparer ConnectionCostComparer = CostComparer.Instance;
-
-		public sealed class CostComparer : IComparer<GraphConnection>
-		{
-			public static readonly CostComparer Instance = new CostComparer();
-			CostComparer() { }
-			public int Compare(GraphConnection x, GraphConnection y)
-			{
-				return x.Cost.CompareTo(y.Cost);
-			}
-		}
-
-		public readonly CPos Destination;
-		public readonly int Cost;
-
-		public GraphConnection(CPos destination, int cost)
-		{
-			Destination = destination;
-			Cost = cost;
-		}
-	}
-
-	sealed class PathGraph : IGraph<CellInfo>
+	sealed class PathGraph : IGraph
 	{
 		public const int PathCostForInvalidPath = int.MaxValue;
 		public const short MovementCostForUnreachableCell = short.MaxValue;
+		const int LaneBiasCost = 1;
 
-		public Actor Actor { get; private set; }
-		public World World { get; private set; }
-		public Func<CPos, bool> CustomBlock { get; set; }
-		public Func<CPos, int> CustomCost { get; set; }
-		public int LaneBias { get; set; }
-		public bool InReverse { get; set; }
-		public Actor IgnoreActor { get; set; }
-
-		readonly BlockedByActor checkConditions;
+		readonly ICustomMovementLayer[] cmls;
```

Comment:
```suggestion
		readonly ICustomMovementLayer[] customLayers;
```
?

Reply:
Renamed usages to `customMovementLayers` everywhere.